test/src/tests-serverModel.ts

```diff
@@ -0,0 +1,1675 @@
+                'UserAttributeChanges'
+            ).to.eql({
+                UserAttributeName: '$Age',
+                New: '42',
+                Old: '37',
+                Deleted: false,
+                IsNewAttribute: false,
+            });
+
+            expect(
+                actualEventObject.UserIdentityChanges,
+                'UserrIdentityChanges'
+            ).to.eql({
+                New: {
+                    IdentityType: Types.IdentityType.Other2,
+                    Identity: 'new_identity',
+                    Timestamp: 1668194307,
+                    CreatedThisBatch: false,
+                },
+                Old: {
+                    IdentityType: Types.IdentityType.Other,
+                    Identity: 'old_identity',
+                    Timestamp: 1668193307,
+                    CreatedThisBatch: false,
+                },
+            });
+            expect(actualEventObject.Store, 'Store').to.eql({});
+            expect(actualEventObject.SDKVersion, 'SDKVersion').to.equal(
+                Constants.sdkVersion
+            );
+            expect(actualEventObject.SessionId, 'SessionId').to.equal(
+                mPStore.sessionId
+            );
+            expect(
+                actualEventObject.SessionStartDate,
+                'SessionStartDate'
+            ).to.equal(mPStore.sessionStartDate.getTime());
+            expect(actualEventObject.Debug, 'Debug').to.equal(false);
+            expect(actualEventObject.Location, 'Location').to.equal(null);
+            expect(actualEventObject.OptOut, 'OptOut').to.equal(null);
+            expect(
+                actualEventObject.ExpandedEventCount,
+                'ExpandedEventCount'
+            ).to.equal(0);
+            expect(actualEventObject.AppVersion, 'AppVersion').to.equal(
+                undefined
+            );
+            expect(actualEventObject.AppName, 'AppName').to.equal(undefined);
+            expect(actualEventObject.Package, 'Package').to.equal(undefined);
+            expect(
+                actualEventObject.ClientGeneratedId,
+                'ClientGeneratedId'
+            ).to.equal(mPStore.clientId);
+            expect(actualEventObject.DeviceId, 'DeviceId').to.equal(
+                mPStore.deviceId
+            );
+            expect(
+                actualEventObject.IntegrationAttributes,
+                'IntegrationAttributes'
+            ).to.eql({});
+
+            // TODO: Should this default to USD?
+            expect(actualEventObject.CurrencyCode, 'CurrencyCode').to.equal(
+                null
+            );
+            expect(actualEventObject.DataPlan, 'DataPlan').to.eql({});
+            expect(actualEventObject.Timestamp, 'Timestamp').to.equal(
+                mPStore.dateLastEventSent.getTime()
+            );
+        });
+
+        it('should create an event object with a user', () => {
+            const event: BaseEvent = {
```

Comment:
I think we should have a TODO to pull these sample BaseEvents and sample Users into it's own file so it can be imported and more DRY.